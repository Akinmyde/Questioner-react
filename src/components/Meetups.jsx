import React, { Component } from 'react';
import Card from './common/Card';
import { checkIsAdmin } from './services/authService';
import { getAllMeetups } from './services/meetupService';
import Loader from './common/Loader';
import exceptionHandler from '../helpers/exceptionHandler';

class Meetups extends Component {
  constructor(props) {
    super(props);
    this.state = { meetups: [], isAdmin: false, loading: false };
  }

  async componentDidMount() {
    this.setState({ loading: true });
    try {
      const isAdmin = checkIsAdmin();
      const meetups = await getAllMeetups();
      this.setState({ meetups, isAdmin });
    } catch (ex) {
      exceptionHandler(ex);
    } finally {
      this.setState({ loading: false });
    }
  }

  render() {
    const { meetups, isAdmin, loading } = this.state;

    return (
      <React.Fragment>
        {loading && <Loader />}
        <div className="flex meetup-card">
          {meetups.map(meetup => (
            <Card
              key={meetup.id}
              id={meetup.id}
              location={meetup.location}
              date={new Date(meetup.happeningon).toDateString()}
              title={meetup.topic}
              imageUrl={meetup.images[0]}
              isAdmin={isAdmin}
            />
          ))}
        </div>
      </React.Fragment>
    );
  }
}

export default Meetups;
