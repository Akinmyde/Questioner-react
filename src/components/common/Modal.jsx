import React from 'react';

const Modal = ({ message, onDelete, onCancel }) => {
  return (
    <div className="modal show-modal">
      <div className="modal-content">
        <span className="close-button" onClick={onCancel}>&times;</span>
        <p className="msg">{message}</p>
        <div>
          <button className="btn-modal btn-yes" onClick={onDelete}>Yes</button>
          <button className="btn-modal btn-no" onClick={onCancel}>No</button>
        </div>
      </div>
    </div>
  );
}
 
export default Modal;