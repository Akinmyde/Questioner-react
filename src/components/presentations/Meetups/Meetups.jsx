import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Loader from '../../common/Loader';
import Modal from '../../common/Modal';

class Meetups extends Component {
  constructor(props) {
    super(props);
    this.state = { showModal: false, currentId: null };
  }

  async componentDidMount() {
    const { getAllMeetups } = this.props;
    await getAllMeetups();
  }

  onDelete = async (meetupId) => {
    const { deleteMeetup } = this.props;
    await deleteMeetup(meetupId);
    this.setState({ showModal: false });
  };

  onDeleteClick = (id) => {
    this.setState({ showModal: true, currentId: id });
  }

  onCancel = () => {
    this.setState({ showModal: false });
  }

  render() {
    const { showModal, currentId } = this.state;
    const { auth, LoadingReducer, meetups } = this.props;
    const { isAdmin } = auth;
    const { loader } = LoadingReducer;
    const { meetupsData } = meetups;
    return (
      <React.Fragment>
        {showModal && <Modal
          message='are you sure you want to delete'
          onDelete={() => {this.onDelete(currentId)}}
          onCancel={this.onCancel}
        /> }
        {loader && <Loader />}
        <div className="flex meetup-card">
          {meetupsData.map(meetup => (
            <div className="meetup-link" key={meetup.id}>
              <div>
                <img className="image" src={meetup.images[0]} alt="logo" />
                <h4><Link className="meetup-link" to={`meetups/${meetup.id}`}>{meetup.topic}</Link></h4>
                <h6>{meetup.location}</h6>
                <span className="text-holder">
                  <ul className="details">
                    <li>{new Date(meetup.happeningon).toDateString()}</li>
                    {isAdmin && (
                      <React.Fragment>
                        <li title="delete"><Link onClick={() => {this.onDeleteClick(meetup.id)}} className="delete"><i className="fas fa-trash" /></Link></li>
                      </React.Fragment>
                    )}
                  </ul>
                </span>
              </div>
            </div>
          ))}
        </div>
      </React.Fragment>
    );
  }
}

export default Meetups;
