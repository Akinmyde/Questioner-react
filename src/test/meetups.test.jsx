import 'babel-polyfill';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Meetups from '../components/presentations/Meetups/Meetups';
import Modal from '../components/common/Modal';

describe('Login Page', () => {
  const props = { meetups: { meetupsData: [] }, LoadingReducer: { loader: false }, auth: { isAdmin: false }, showModal: true, currentId: 1 };
  it('should render meetups correctly', async () => {
    const wrapper = shallow(<Meetups { ...props }/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().onCancel());
    expect(wrapper.instance().onDeleteClick(1));
    expect(wrapper.instance().onDelete(1));
  });
});
