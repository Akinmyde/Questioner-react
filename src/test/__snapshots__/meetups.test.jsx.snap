// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetups Page should match snapshot componentDidMount 1`] = `Promise {}`;

exports[`Meetups Page should render signup correctly 1`] = `
<Fragment>
  <Loader />
  <div
    className="flex meetup-card"
  />
</Fragment>
`;
